### PR TITLE
add gcd test for DateTime

### PR DESCRIPTION
**Pull request by PSeitz**
add gcd test for DateTime

### CODE REVIEW

File: fastfield_codecs/src/linearinterpol.rs [hunk 1]
```diff
@@ -102,7 +102,11 @@ fn get_slope(first_val: u64, last_val: u64, num_vals: u64) -> f32 {
 
 #[inline]
 fn get_calculated_value(first_val: u64, pos: u64, slope: f32) -> u64 {
-    first_val + (pos as f32 * slope) as u64
+    if slope < 0.0 {
+        (first_val - (pos as f32 * slope) as u64) as u64
```
Comment (fulmicoton) [inline fastfield_codecs/src/linearinterpol.rs]: i don't understand... why would that be correct?

Comment (PSeitz) [reply fastfield_codecs/src/linearinterpol.rs]: It's not, it should be
`(pos as f32 * -slope)`

Comment (PSeitz) [reply fastfield_codecs/src/linearinterpol.rs]: I moved this to the bugfix PR https://github.com/quickwit-oss/tantivy/pull/1439
This PR is now only adding a test and exposing the gcd

Comment (fulmicoton) [reply fastfield_codecs/src/linearinterpol.rs]: Why is it * -slope?

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Let's not expose `gcd` and just check on the compressed len in the unit test...

File: src/fastfield/gcd.rs [hunk 2]
```diff
@@ -14,10 +14,11 @@ pub const GCD_CODEC_ID: u8 = 4;
 /// Holds the data and the codec to the read the data.
 #[derive(Clone)]
 pub struct GCDFastFieldCodec<CodecReader> {
-    gcd: u64,
+    pub(crate) gcd: u64,
```
Comment (fulmicoton) [inline src/fastfield/gcd.rs]:
```suggestion
    gcd: u64,
```

File: src/fastfield/gcd.rs [hunk 3]
```diff
@@ -228,6 +229,76 @@ mod tests {
...
+        let mut schema_builder = Schema::builder();
+        let date_options = DateOptions::default()
+            .set_fast(Cardinality::SingleValue)
+            .set_precision(precision);
+        let field = schema_builder.add_date_field("field", date_options);
+        let schema = schema_builder.build();
+
+        let docs = vec![doc!(field=>time1), doc!(field=>time2), doc!(field=>time3)];
+
+        let directory = get_index(&docs, &schema, codec_name.clone().into())?;
```
Comment (fulmicoton) [inline src/fastfield/gcd.rs]:
```suggestion
        let directory = get_index(&docs, &schema, codec_name.into())?;
```

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**